### PR TITLE
Adding protocol and policyactions definitions

### PR DESCRIPTION
**Pull request by brahmaroutu**
Added Protocol and policyActionsConfigMap to types as per spec definitions

### CODE REVIEW

**Review by rrati — APPROVED** [review]
lgtm

File: crds/objectstorage.k8s.io_bucketaccessclasses.yaml [hunk 1]
```diff
@@ -36,8 +36,18 @@ spec:
...
-            type: string
+          policyActionsConfigMap:
+            properties:
+              name:
+                description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names'
+                type: string
+              namespace:
+                description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/'
+                type: string
+              uid:
```
Comment (jeffvance) [inline crds/objectstorage.k8s.io_bucketaccessclasses.yaml:47]: Is uid needed?

File: crds/objectstorage.k8s.io_bucketrequests.yaml [hunk 2]
```diff
@@ -42,48 +42,16 @@ spec:
...
-                        type: string
-                      signatureVersion:
-                        enum:
-                        - s3v2
-                        - s3v4
-                        type: string
-                      version:
-                        type: string
-                    type: object
+                  version:
```
Comment (jeffvance) [inline crds/objectstorage.k8s.io_bucketrequests.yaml:51]: Does it matter that the API defines this as "azureBlob" on L48?

File: crds/objectstorage.k8s.io_buckets.yaml [hunk 3]
```diff
@@ -88,12 +88,19 @@ spec:
...
-                    - s3
-                    - azure
-                    - gcs
-                    type: string
+                  requestedProtocol:
+                    properties:
+                      name:
+                        enum:
+                        - s3
+                        - azure
```
Comment (jeffvance) [inline crds/objectstorage.k8s.io_buckets.yaml]: ditto on "azure" vs "azureBlob"?

**Comment by jeffvance** [comment]
a couple of minor comments, but lgtm

File: crds/objectstorage.k8s.io_bucketaccessclasses.yaml (hunk 1 above)
Comment (brahmaroutu) [reply crds/objectstorage.k8s.io_bucketaccessclasses.yaml:47]: not required but I am reusing the ObjectReference to not define one more type.

File: crds/objectstorage.k8s.io_bucketrequests.yaml (hunk 2 above)
Comment (brahmaroutu) [reply crds/objectstorage.k8s.io_bucketrequests.yaml:51]: @rrati This is how you named, not sure of the context. 'protocol: azure' and the struct that container details protocol struct named as 'azureBlob'.

File: crds/objectstorage.k8s.io_buckets.yaml (hunk 3 above)
Comment (brahmaroutu) [reply crds/objectstorage.k8s.io_buckets.yaml]: @jeffvance  thanks for the review, changed enum to azureBlob

Comment (brahmaroutu) [reply crds/objectstorage.k8s.io_buckets.yaml]: thanks for the review, changed enum to azureBlob

**Pull request merged**